### PR TITLE
[Select] Create options onPaste event

### PR DESCRIPTION
**Pull request by SerhiiTsybulskyi**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

https://github.com/reaviz/reablocks/assets/5399389/033f4f36-02bc-4580-a7b9-c509237d1832

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/form/Select/utils/keyboard.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
+export const keyNameToCode = {
```
Comment (SerhiiTsybulskyi) [inline src/form/Select/utils/keyboard.ts:1]: I've added this mapping to keep readability for 
```
selectOnKeys={['Enter', 'Space', 'Comma']
```
I think this more friendly format instead array of numbers (key codes)

Comment (amcdnl) [reply src/form/Select/utils/keyboard.ts:1]: You don't really need this with https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key

File: src/form/Select/Select.tsx [hunk 2]
```diff
@@ -638,6 +637,45 @@ export const Select: FC<Partial<SelectProps>> = ({
     [createable, menuDisabled, onInputBlur, toggleSelectedOption]
   );
 
+  const onPasteHandler = useCallback(
+    (e: React.ClipboardEvent<HTMLInputElement>) => {
+      if (createable && multiple) {
```
Comment (amcdnl) [inline src/form/Select/Select.tsx:642]: This needs to be an option `createOnPaste` or `selectOnPaste` might actually be more correct.

File: src/form/Select/utils/keyboard.ts (hunk 1 above)
Comment (SerhiiTsybulskyi) [reply src/form/Select/utils/keyboard.ts:1]: This is works for Keyboard events (manually filling) but didn't work on paste due to ClipboardEvent (another way to interact with input needs a little bit different handler)

File: src/form/Select/Select.tsx [hunk 3]
```diff
@@ -639,7 +645,7 @@ export const Select: FC<Partial<SelectProps>> = ({
 
   const onPasteHandler = useCallback(
     (e: React.ClipboardEvent<HTMLInputElement>) => {
-      if (createable && multiple) {
+      if (createOnPaste && multiple) {
```
Comment (amcdnl) [inline src/form/Select/Select.tsx:648]: Does it select if its already a 'created' option?

Comment (SerhiiTsybulskyi) [reply src/form/Select/Select.tsx:648]: yes

Comment (amcdnl) [reply src/form/Select/Select.tsx:648]: if creatable = false but I paste in a option this condition is invalid. I think it should be `selectOnPaste` then when combined w/ `creatable` would create them or of not just select them.

**Review by amcdnl — APPROVED** [review]

**Pull request merged**